Move travis to 18.04

Karmastic opened the pull request:
We want to start building and testing devnet (nightly) on Ubuntu 18.04 so we can move production to it after more testing.
Remove building of debian package - it's now done for release builds only.
Remove GATE_PREFIX - we'll use separate buckets for the same purpose.

NOTE: This will fail to install and run on CentOS due to an obsolete glibc version in CentOS.

Commit:
Move travis to 18.04

We want to start building and testing devnet (nightly) on Ubuntu 18.04 so we can move production to it after more testing.
Remove building of debian package - it's now done for release builds only.
Remove GATE_PREFIX - we'll use separate buckets for the same purpose.

algobolson commented on scripts/build_packages.sh:
```diff
@@ -106,30 +99,16 @@ for var in "${VARIATION_ARRAY[@]}"; do
-        tar -zcf ${PKG_ROOT}/${GATE_PREFIX}tools_${CHANNEL}_${PKG_NAME}_${FULLVERSION}.tar.gz * >/dev/null 2>&1
+        tar -zcf ${PKG_ROOT}/tools_${CHANNEL}_${PKG_NAME}_${FULLVERSION}.tar.gz * >/dev/null 2>&1
         popd
-
```
build_release.sh calls build_packages.sh and expects that to build the .deb
we could turn it off here with a variable test, e.g. `if [ ! -z "${BUILD_DEB}" ];` or so

tsachiherman commented:
I think that our goal should be expanding our supported platform base rather than shrinking it..
What would it take to have a "special build" for RedHat/CentOS builds ?

algobolson commented:
ubuntu 16.04 is supposed to get maintenance upgrades through 2020 ( https://ubuntu.com/about/release-cycle ), but _we_ don't have to support it that long. Moving to newer systems makes things easier for us because they have newer versions of things installed with bug fixes and nicer features. The release build should continue to support Ubuntu 18.04 and Centos 7.

Commit:
Make building debian pkg optional

We still use the script in build_packages.sh to generate the deb package for production release builds; make it optional and have those builds enable it.

Karmastic commented:
> I think that our goal should be expanding our supported platform base rather than shrinking it..
> What would it take to have a "special build" for RedHat/CentOS builds ?

I'm going to exercise our RPM package for R1 and R2, our CentOS machines.  They'll be supported - they just won't support the installer (unless the user wants to build glibc for CentOS 7 themselves)

algobolson commented on scripts/build_packages.sh:
```diff
@@ -128,7 +121,7 @@ for var in "${VARIATION_ARRAY[@]}"; do
             fi
             pushd ${DEBTMP}
-	    cp -p *.deb ${PKG_ROOT}/${GATE_PREFIX}algorand_${CHANNEL}_${PKG_NAME}_${FULLVERSION}.deb
+            cp -p *.deb ${PKG_ROOT}/${GATE_PREFIX}algorand_${CHANNEL}_${PKG_NAME}_${FULLVERSION}.deb
```
one more GATE_PREFIX

Commit:
Remove extraneous GATE_PREFIX

Was a nop but doesn't belong anymore

Commit:
Merge remote-tracking branch 'origin/master' into david/moveTo1804

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/algorand/go-algorand?pullRequest=104) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/algorand/go-algorand?pullRequest=104) before we can accept your contribution.<br/><hr/>**David Shoots** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/algorand/go-algorand?pullRequest=104) it.</sub>
